src/mudlet.cpp

```diff
@@ -1293,38 +1293,104 @@ void mudlet::addConsoleForNewHost(Host* pH)
 
 void mudlet::slot_timer_fires()
 {
-    QTimer* pQT = (QTimer*)sender();
+    auto pQT = qobject_cast<QTimer*>(sender());
     if (!pQT) {
         return;
     }
-    if (mTimerMap.contains(pQT)) {
-        TTimer* pTT = mTimerMap[pQT];
-        pTT->execute();
-        if (pTT->checkRestart()) {
-            pTT->start();
+
+    // Search through the Host instances to see which one has a QTimer that is
+    // the one that went off:
+    QMutableMapIterator<Host*, QMap<QTimer*, TTimer*>> itHostTimerMap(mHostTimerMap);
```

Comment:
It doesn't seem that we're modifying the `mHostTimerMap` here, are we? Can we use the faster const iterator?

Reply:
Probably - that is likely a remnant of the previous code that I had not spotted had become constant...

Reply:
Sweet. This path is a hotspot that we must make as efficient as possible :)